bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/handler/GardenaAccountHandler.java

```diff
@@ -51,80 +64,188 @@
+        apiCallSuppressionStart = "".equals(propertyValue) ? Instant.MIN : Instant.parse(propertyValue);
+        propertyValue = properties.getOrDefault(GardenaBindingConstants.API_CALL_SUPPRESSION_END, "");
+        apiCallSuppressionEnd = "".equals(propertyValue) ? Instant.MIN : Instant.parse(propertyValue);
+    }
+
+    /**
+     * Get the duration remaining until the end of the api call suppression window, or Duration.ZERO if we are outside
+     * the call suppression window.
+     *
+     * @return the duration until the end of the suppression window, or zero.
+     */
+    private Duration getApiCallSuppressionDurationRemaining() {
+        Instant now = Instant.now();
+        if (now.isAfter(apiCallSuppressionStart) && now.isBefore(apiCallSuppressionEnd)) {
+            return Duration.between(now, apiCallSuppressionEnd);
+        }
+        return Duration.ZERO;
+    }
+
+    /**
+     * Updates the time when api call suppression begins to now(), and the time when api call suppression ends to now()
+     * plus the given delay. The delay must not be a negative duration. Saves the start and end time values as
+     * properties to ensure consistent behaviour across restarts.
+     *
+     * @param delay the duration between the start and end of the suppression window.
+     */
+    private void apiCallSuppressionPropertiesUpdate(Duration delay) {
+        Duration offset = delay.isNegative() ? Duration.ZERO : delay;
+        apiCallSuppressionStart = Instant.now();
+        apiCallSuppressionEnd = apiCallSuppressionStart.plus(offset);
+        Thing thing = getThing();
+        thing.setProperty(GardenaBindingConstants.API_CALL_SUPPRESSION_START, apiCallSuppressionStart.toString());
+        thing.setProperty(GardenaBindingConstants.API_CALL_SUPPRESSION_END, apiCallSuppressionEnd.toString());
     }
 
     @Override
     public void initialize() {
         logger.debug("Initializing Gardena account '{}'", getThing().getUID().getId());
-        initializeGardena();
+        loadApiCallSuppressionProperties();
+        Duration delay = getApiCallSuppressionDurationRemaining();
+        if (Duration.ZERO.equals(delay)) {
+            // do immediate initialisation
+            scheduler.submit(() -> initializeGardena());
+        } else {
+            // delay the initialisation
+            scheduleReinitialize(delay);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, getUiText());
+        }
     }
 
     public void setDiscoveryService(GardenaDeviceDiscoveryService discoveryService) {
         this.discoveryService = discoveryService;
     }
 
+    /**
+     * Format a localised description text to display on the thing main UI page.
+     *
+     * @return the (localised) description text.
+     */
+    private String getUiText() {
+        String template = i18nProvider.getText(bundle, RECONNECT_MSG_KEY, RECONNECT_MSG_KEY,
```

Comment:
This is not needed, `updateStatus()` supports "@text/" references directly, i.e. "@text/accounthandler.waiting-to-reconnect-at-time". Example:
https://github.com/openhab/openhab-addons/blob/67006de60663fc367a61a8396f9a1b82543a7709/bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/handler/DanfossAirUnitHandler.java#L110-L111

Reply:
What's the 'this' in that statement though? ;-) The indirection via string constant?

Reply:
Direct usage of `LocaleProvider` and `TranslationProvider`.

Reply:
Thanks for the tip. However if you look at the request from @maniac103 above, he requested to have different formatting depending on whether the call suppression expires today (show the time only) or tomorrow (show the date and the time). So I do need a separate method for this.

Reply:
I don't really understand what you mean? You have one i18n text and one parameter which you can format any way you prefer. However, this custom implementation is unneeded and overly complex when you can simply use a "@text/" reference for `updateStatus()`?

Reply:
^
Two things..
1. the code formatting is actually quite complex, so it would be hard to write the formatting as a "one liner"
2. updateStatus() is called in more than one part of the code, so the "hard one liner" would be duplicated.

Reply:
Latest version of the code is below.
If you want to convert that to a one liner, you are welcome to try it :)
```
private String getUiText() {
    Instant until = apiCallSuppressionUntil;
    if (until != null) {
        ZoneId zone = timeZoneProvider.getTimeZone();
        boolean isToday = LocalDate.now(zone).equals(LocalDate.ofInstant(until, zone));
        DateTimeFormatter formatter = isToday ? DateTimeFormatter.ofLocalizedTime(FormatStyle.MEDIUM)
                : DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM);
        String value = i18nProvider.getText(bundle, RECONNECT_MSG_KEY, null, localeProvider.getLocale(),
                formatter.format(ZonedDateTime.ofInstant(until, zone)));
        return value != null ? value : "";
    }
    return "";
}
```

Reply:
Almost forgot about this one. I was suggesting to use a "@text/" reference for `updateStatus()` as it directly supports it. I was not suggesting a one-liner or to introduce redundancy. But I'll still take the challenge and write the code to express my suggestion, will post as separate suggestion.